sdk/clientcore/core/src/main/java/io/clientcore/core/utils/CoreUtils.java

```diff
@@ -432,19 +431,36 @@ public static SerializationFormat serializationFormatFromContentType(HttpHeaders
      * @param url the base URL to which the query parameter will be appended.
      * @param key the name of the query parameter (e.g., "api-version", "name", "After").
      * @param value the value of the query parameter
+     * @param delimiter The delimiter to use if the value is a list.
      * @return the updated URL with the appended query parameter.
      */
-    public static String appendQueryParam(String url, String key, String value) {
-        if (value == null) {
+    public static String appendQueryParam(String url, String key, Object value, char delimiter) {
```

Comment:
Instead of changing this existing API, could we create a new API? Ex, `appendMultiQueryParam(String url, String key, List<String> values)`

Or maybe better long-term, have the annotation processor generated code use `UriBuilder` which is more capable of handling this.

Reply:
Also, this API doesn't support whether the value for the query param was already encoded, which we'll need.

Reply:
yeah, updated the design to have a QueryParameter model so that we can extend it to future needs too of supporting encoded etc.